### ISSUE TITLE
pyright 1.1.100 causes type errors with Django models

### ISSUE THREAD

**Issue by ChristopherZhong** [microsoft/pyright#1359]
**Describe the bug**

Pyright 1.1.100 causes some errors with Django models.
The following is an example that can cause the errors.

```python
from django.db import models

# Create your models here.
class ExampleModel(models.Model):
    test_field = models.CharField()

    def test_concat(self):
        return self.test_field + 'some text'

    def test_size(self):
        return ExampleModel.objects.count()
```

The following are the errors using `npx pyright`.

```zsh
(venv) ➜  test-django-app-3.1.5 npx pyright
npx: installed 1 in 1.647s
No configuration file found.
stubPath ~/test-django-app-3.1.5/typings is not a valid directory.
Assuming Python platform Darwin
Searching for source files
Auto-excluding ~/test-django-app-3.1.5/venv
Found 13 source files
~/test-django-app-3.1/sample_app/models.py
  8:16 - error: Operator "+" not supported for types "CharField" and "Literal['some text']"
    Operator "+" not supported for types "CharField" and "Literal['some text']" (reportGeneralTypeIssues)
  11:29 - error: Cannot access member "objects" for type "Type[ExampleModel]"
    Member "objects" is unknown (reportGeneralTypeIssues)
2 errors, 0 warnings, 0 infos 
Completed in 1.06sec
```

**To Reproduce**
Steps to reproduce the behavior.

```zsh
➜  ~ mkdir test-django-app-3.1.5
➜  ~ cd test-django-app-3.1.5
➜  test-django-app-3.1.5 python3 -m venv venv
➜  test-django-app-3.1.5 source venv/bin/activate
(venv) ➜  test-django-app-3.1.5 pip install Django==3.1.5 
Collecting Django==3.1.5
  Using cached https://files.pythonhosted.org/packages/b2/8f/d27f35f0639103271231bc81a96ad9188e6b5bc878e140ccb0dc610ccef0/Django-3.1.5-py3-none-any.whl
Collecting sqlparse>=0.2.2 (from Django==3.1.5)
  Using cached https://files.pythonhosted.org/packages/14/05/6e8eb62ca685b10e34051a80d7ea94b7137369d8c0be5c3b9d9b6e3f5dae/sqlparse-0.4.1-py3-none-any.whl
Collecting pytz (from Django==3.1.5)
  Using cached https://files.pythonhosted.org/packages/89/06/2c2d3034b4d6bf22f2a4ae546d16925898658a33b4400cfb7e2c1e2871a3/pytz-2020.5-py2.py3-none-any.whl
Collecting asgiref<4,>=3.2.10 (from Django==3.1.5)
  Using cached https://files.pythonhosted.org/packages/89/49/5531992efc62f9c6d08a7199dc31176c8c60f7b2548c6ef245f96f29d0d9/asgiref-3.3.1-py3-none-any.whl
Installing collected packages: sqlparse, pytz, asgiref, Django
Successfully installed Django-3.1.5 asgiref-3.3.1 pytz-2020.5 sqlparse-0.4.1
WARNING: You are using pip version 19.2.3, however version 20.3.3 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
(venv) ➜  test-django-app-3.1.5 django-admin startproject django_app .
(venv) ➜  test-django-app-3.1.5 django-admin startapp example_app
```
Note: I changed the absolute path to a relative path `~`.

Then copy the contents of the above code into the `test-django-app-3.1.5/example_app/models.py` file.

**Expected behavior**

There should be no errors.
The following shows the expected output.

```zsh
(venv) ➜  test-django-app-3.1.5 npx pyright@1.1.99
npx: installed 1 in 1.436s
No configuration file found.
stubPath ~/test-django-app-3.1.5/typings is not a valid directory.
Assuming Python platform Darwin
Searching for source files
Auto-excluding ~/test-django-app-3.1.5/venv
Found 13 source files
0 errors, 0 warnings, 0 infos 
Completed in 0.658sec
```

Note: I changed the absolute path to a relative path `~`.

**Additional context**

I'm not sure if this is important but the python version is `3.8.2`.

```zsh
(venv) ➜  test-django-app-3.1.5 python --version
Python 3.8.2
```

**Comment by erictraut** [comment]
Do you have the [django type stubs](https://pypi.org/project/django-stubs/) installed?

Are you seeing a different behavior with Pyright 1.1.100 than with previous versions?

I have django stubs installed, and I'm not able to repro the problem you're reporting.

**Comment by erictraut** [comment]
I suspect this is due to a regression I introduced in 1.1.100. I've fixed the problem and published 1.1.101. Please try to repro your problem with this version. If it no longer happens, feel free to close the issue.

**Comment by ChristopherZhong** [comment]
> Do you have the [django type stubs](https://pypi.org/project/django-stubs/) installed?

Django type stubs are not installed.

> Are you seeing a different behavior with Pyright 1.1.100 than with previous versions?

Yes. In Pyright 1.1.99 and below, the output is the same as the expected behavior.

I'll start testing it with the new version.

**Comment by ChristopherZhong** [comment]
Thank you. Issue is resolved

**Comment by crazyoptimist** [comment]
Thank  you @erictraut

**Comment by erictraut** [comment]
@shughes-uk, if you think you are seeing a bug or a regression, please open a new issue and provide repro steps, preferably with a minimal, self-contained code sample.
